cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/executor/DatasetAdminService.java

```diff
@@ -110,6 +141,9 @@ public void drop(Id.DatasetInstance datasetInstanceId, DatasetTypeMeta typeMeta,
       DatasetAdmin admin = type.getAdmin(DatasetContext.from(datasetInstanceId.getNamespaceId()), spec);
       admin.drop();
     }
+
+    // Remove metadata for the dataset
+    metadataStore.removeMetadata(datasetInstanceId);
```

Comment:
Since we are only deleting the metadata for the dataset, we don't need `DatasetClasLoaderProvider` to be open.